runner/runner.go

```diff
@@ -0,0 +1,316 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright (C) 2022 Renesas Electronics Corporation.
+// Copyright (C) 2022 EPAM Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aoscloud/aos_common/aoserrors"
+	"github.com/aoscloud/aos_common/api/cloudprotocol"
+	"github.com/coreos/go-systemd/v22/dbus"
+	log "github.com/sirupsen/logrus"
+)
+
+/***********************************************************************************************************************
+ * Consts
+ **********************************************************************************************************************/
+
+const unitStatusChannelSize = 10
+
+const (
+	defaultStartTimeout    = 30
+	startTimeoutMultiplier = 2
+)
+
+const systemdUnitNameTamplate = "aos-service@%s.service"
+
+const errNotLoaded = "not loaded"
+
+const jobStateDone = "done"
+
+/***********************************************************************************************************************
+ * Vars
+ **********************************************************************************************************************/
+
+var ErrStartInstance = errors.New("can't start service instance")
+
+/***********************************************************************************************************************
+  Types
+ **********************************************************************************************************************/
+
+// StartInstanceParams start instance parameters.
+type StartInstanceParams struct {
+	StartInterval   time.Duration
+	StartBurst      uint
+	RestartInterval time.Duration
+}
+
+// InstanceStatus service instance status.
+type InstanceStatus struct {
+	InstanceID string
+	State      string
+	Err        error
+	ExitCode   int
+}
+
+// Runner runner instance.
+type Runner struct {
+	systemd            *dbus.Conn
+	instanceStatusChan chan []InstanceStatus
+	startingUnits      map[string]chan dbus.UnitStatus
+	runningUnits       map[string]bool
+	stopChan           chan bool
+	sync.RWMutex
+}
+
+/***********************************************************************************************************************
+  Public
+ **********************************************************************************************************************/
+
+// New creates new systemd runner.
+func New() (runner *Runner, err error) {
```

Comment:
You can remove named values from return.

Reply:
in this particular case prefer to have named return values